clang/lib/Sema/SemaBounds.cpp

```diff
@@ -1919,6 +1960,13 @@ namespace {
   // Otherwise an expression denotes an rvalue.
 
   public:
+    BoundsExpr *Check(Stmt *S, CheckedScopeSpecifier CSS) {
+      EquivExprSets EQ;
+      EquivExprSets *UEQ = new EquivExprSets();
+      EqualExprTy *G = new EqualExprTy();
```

Comment:
The logic for updating UEQ and G in other methods is more complicated than CheckDeclRefExpr, and EQ, UEQ, and G are not always handled together. For example, CheckBinaryOperator will need to make these calls for assignment operators:
```
Let UEQ_LHS, G_LHS
Check(EQ, UEQ_LHS, G_LHS)
Let UEQ_RHS, G_RHS
Check(UEQ_LHS, UEQ_RHS, G_RHS)
```
The last Check call uses the EQ set that resulted in checking the left-hand side of the assignment to update the UEQ and G sets for the right-hand side of the assignment. I don't know that this lends itself well to passing around a single class containing EQ, UEQ, and G. It might be possible to wrap UEQ and G into a class; I can look into this further.